src/usecase/tests/address_ucase_test.go

```diff
@@ -0,0 +1,334 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/Aiscom-LLC/meals-api/src/delivery"
+	"github.com/Aiscom-LLC/meals-api/src/delivery/middleware"
+	"github.com/Aiscom-LLC/meals-api/src/repository"
+	"github.com/appleboy/gofight/v2"
+	"github.com/buger/jsonparser"
+	"github.com/go-playground/assert/v2"
+	"net/http"
+	"testing"
+)
+
+func TestAddAddress(t *testing.T) {
+	r := gofight.New()
+
+	var clientRepo = repository.NewClientRepo()
+	var userRepo = repository.NewUserRepo()
+	userResult, _ := userRepo.GetByKey("email", "admin@meals.com")
+	jwt, _, _ := middleware.Passport().TokenGenerator(&middleware.UserID{userResult.ID.String()})
+	clientResult, _ := clientRepo.GetByKey("name", "Dymi")
+	clientID := clientResult.ID.String()
+
+	// Trying to creat new address
+	// Should be success
+	r.POST("/clients/"+clientID+"/addresses").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"city":   "Sochi",
+			"street": "Gagar",
+			"house":  "31",
+			"floor":  2,
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusOK, r.Code)
+		})
+
+	// Trying to creat new address with invalid or not full data
+	// Should return an error
+	r.POST("/clients/"+clientID+"/addresses").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"city": "Sochi",
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusBadRequest, r.Code)
+		})
+
+	// Trying to creat new address on non-valid client ID
+	// Should return an error
+	r.POST("/clients/qwerty/addresses").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"city":   "Sochi",
+			"street": "Gagar",
+			"house":  "31",
+			"floor":  2,
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusBadRequest, r.Code)
+		})
+
+	// Trying to creat new address without authorize
```

Comment:
create, and if you're not authorizing why would you need `SetCookie`? could be removed